pkg/nsx/services/ippool/builder.go

```diff
@@ -92,15 +92,17 @@ func (service *IPPoolService) buildIPSubnetIntentPath(IPPool *v1alpha2.IPPool, s
 }
 
 func (service *IPPoolService) buildIPSubnet(IPPool *v1alpha2.IPPool, subnetRequest v1alpha2.SubnetRequest) *model.IpAddressPoolBlockSubnet {
-	// TODO: Get the IPBlockPath by IPPool's namespace, external and private
-	IpBlockPath := String("/infra/ip-blocks/block-test")
+	IpBlockPath := String("")
+	VPCInfo := commonctl.ServiceMediator.GetVPCsByNamespace(IPPool.Namespace)
+	if len(VPCInfo) == 0 {
+		log.Error(nil, "failed to find VPCInfo for IPPool CR", "IPPool", IPPool.Name, "namespace %s", IPPool.Namespace)
+		return nil
+	}
+
 	if IPPool.Spec.Type == common.IPPoolTypePrivate {
-		VPCInfo := commonctl.ServiceMediator.ListVPCInfo(IPPool.Namespace)
-		if len(VPCInfo) == 0 {
-			return nil
-		}
-		// TODO: Get the IPBlockPath by IPPool's namespace, external and private
-		IpBlockPath = String(fmt.Sprintf("/orgs/%s/projects/%s/infra/ip-blocks/block-test", VPCInfo[0].OrgID, VPCInfo[0].ProjectID))
+		IpBlockPath = String(VPCInfo[0].PrivateIpv4Blocks[0])
```

Comment:
For IP exhausted case, we need to return an available ip block instead of the first one

Reply:
agree

Reply:
would fix it after tao's patch merged.

Reply:
Add TODO for this one